### PR TITLE
Correct layering violations related to the proof property

### PR DESCRIPTION
**Pull request by selfissued**
Remove uses of `proof` where a specific securing mechanism is not being used or described.  Where `proof` is being used for Data Integrity, be explicit that that is the case.

This will resolve most of https://github.com/w3c/vc-data-model/issues/1377.

cc: @jyasskin

***
<a href="https://pr-preview.s3.amazonaws.com/selfissued/vc-data-model/pull/1397.html" title="Last updated on Jan 12, 2024, 9:24 PM UTC (ca11418)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1397/611dafd...selfissued:ca11418.html" title="Last updated on Jan 12, 2024, 9:24 PM UTC (ca11418)">Diff</a>

### CODE REVIEW

**Review by decentralgabe — APPROVED** [review]

**Review by OR13 — APPROVED** [review]

**Review by msporny — CHANGES_REQUESTED** [review]
This PR removes the `proof` property as an extension point, which we don't have consensus to do. It also makes a number of changes that are already made in other PRs, such as removing `proof` from a few examples. In many cases, it edits text that is in the process of being removed in other PRs.

I suggest waiting for the other PRs to settle and then seeing what is left. Some of the text in this PR is ok, but the removal of `proof` entirely as an extension point is the biggest issue.

**Comment by selfissued** [comment]
@msporny can you propose specific wording changes to the PR to address your concern?  I'm OK leaving `proof` as an extension point provided that we leave it up to specifications using it to define how it is used.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-12-19-vcwg-special#section2-1) on 2023-12-19

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.1. Correct layering violations related to the proof property (pr vc-data-model#1397)

_See github pull request [vc-data-model#1397](https://github.com/w3c/vc-data-model/pull/1397)._




**Brent Zundel:** Mike, you opened this one about an hour ago, I don't imagine people have had time to review.  

**Michael Jones:** Thank you.  
… I was assigned an issue about working on correcting layering violations. I finally allocated a couple days time to work on VCDM. Yay -- to try and help us finish.  
… Most of the layering violations that were present that I found had to do with unlabeled uses of the `proof` property. Places where `proof` was present but it wasn't said that it was specific to certain securing mechanisms. I did some of this a while ago to sync with vc-jose-cose.  
… This PR tries to be even handed around where `proof` is used and we say it's an example of using Data Integrity and where it wasn't needed I just removed it.  
… There is another PR to go along with this in Data Integrity where I wanted to be really careful where if I removed anything in the VCDM PR that needed to be said in DI that I wanted to be sure that DI said it.  
… The good news is that I read every line of my PR that was being removed and looked at DI and made sure that in almost all cases it was already said in DI. Only one sentence wasn't there. I am trying to be even-handed and positive and the point is to be agnostic to the securing mechanisms.  
… I hope that this helps close the issue about having an even-handed verification method.  
… Unless there are questions from people who have actually read it, I suggest we move on. I just wanted to flag to say we want people to read it because it gets us to closer to CR.  

**Brent Zundel:** If anyone has read it please jump on the queue.  

**Manu Sporny:** I was able to read it before the call. Some of the changes in the text being made has already been removed in other PRs.  
… That includes removing `proof` from some examples that has already been done in other PRs. That includes removing conformance statements around `proof` that are also gone. This PR would be removing things that are already taken care of elsewhere.  
… The one problem is with removing `proof` as an extension point. There's a mismatch in understanding there I think. The `proof` property is not meant to be DI-only, but an extension point to be used by other securing mechanisms, some that may not be DI.  
… That's my concern with the PR -- where it removes the extension point.  
… I don't think that was the intention of `proof` being an extension point. Otherwise, the PR is removing things that other PRs remove and that's fine. So I'm only worried about removing `proof` as an extension point.  

**Michael Jones:** Glad to hear that other PRs are already handling much of this.  
… We can all declare victory there.  
… With respect to the extension point point ... I tried to be really judicious about whether there are examples using `proof` ... where there are examples I said this is an example that could be secured using DI.  
… Quoting myself, something to the effect "`proof` can be included if specified for use by a securing mechanism." So any securing mechanism can be used, but it's not something you'd put in if the securing mechanism doesn't call it out. So I don't see it as needing to be a general extension point.  
… Any securing mechanism can define an extension point and one of those can be `proof`.  
… We can talk about that in the PR if you want.  

**Brent Zundel:** That's my recommendation, continue the conversation in the PR. It's been live for an hour, so get your reviews in.  

**Michael Jones:** Live for an hour, but two days of work, had to read lots of text.  

**Brent Zundel:** Thank you, Mike.  

</details>

**Review by awoie — APPROVED** [review]

File: index.html
Comment (iherman) [inline index.html:805]: I do not know yet how this PR will be reconciled with #1389 (there are many overlapping issues). While I _agree_ with the change proposal here, I would request not to adopt it ***for now***. The reason is that I plan a separate PR, following the discussion in #1135, which would also add a JWT-based diagram alongside the current one, and which, therefore, makes a lot of changes on this text. The updated text (which is already on my machine) will (already does) include this type of remark.

I plan to submit the related PR once the current, pre-CR PR-s are finalized (merged or closed).

Thx.

File: index.html
Comment (iherman) [inline index.html:898]: Same comment as above.

File: index.html
Comment (iherman) [inline index.html:2068]: I am afraid the same issue arises as in my comment on https://github.com/w3c/vc-data-model/pull/1389#issuecomment-1860849474. 

_If_ any securing mechanism _MAY_ use a `proof` property, then this property must be formally defined in the VCDM specification. At the moment, because it is formally defined in the DI spec, using `proof` means that a third party securing mechanism, which is ***not*** DI, would have a dependency on the DI spec. That is a different layering violation that we would introduce.

If the `proof` property is formally _defined_ in the VCDM as _one possible way of hooking a securing mechanism into VCDM_ then I believe we are fine (provided the DI spec is also updated accordingly).

I give more LD terms in https://github.com/w3c/vc-data-model/pull/1389#issuecomment-1860849474, but I believe the underlying problem is identical.

File: index.html
Comment (TallTed) [inline index.html:3544]:
```suggestion
information to the securing mechanism used. The <code>evidence</code>
```

File: index.html
Comment (TallTed) [inline index.html:4584]:
```suggestion
<var>inputBytes</var> as an input parameter; parses the value as JSON into
```

File: index.html
Comment (TallTed) [inline index.html:4585]:
```suggestion
an object; removes any `proof` property, if present; and returns the result as a JSON object.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Comment by msporny** [comment]
> @msporny can you propose specific wording changes to the PR to address your concern? I'm OK leaving `proof` as an extension point provided that we leave it up to specifications using it to define how it is used.

Yes, many of the significant concrete change suggestions are in these PRs:

* PR #1389
* PR #1393
* PR #1394 

If/when those PRs land, the other concrete changes I'd suggest to this PR would be in text that is deleted (and therefore, there is no sense in making those changes since the text will disappear if the PRs above are merged).

Most of the changes fall into those two categories. We'll have to see what the delta is if/if not the PRs above land and go from there.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-01-03-vcwg#section1-1) on 2024-01-03

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.1. Correct layering violations related to the proof property (pr vc-data-model#1397)

_See github pull request [vc-data-model#1397](https://github.com/w3c/vc-data-model/pull/1397)._




**Brent Zundel:** looking at 1397 (https://github.com/w3c/vc-data-model/pull/1397) - has been superseded by other PRs? is that correct.  

**Manu Sporny:** next step is for Mike to look at the latest spec and see what the delta is. probably a minimal change set at this point.  

</details>

**Comment by selfissued** [comment]
I plan to resolve merge conflicts after PR #1403 has been merged, and ideally, also #1404.

**Comment by selfissued** [comment]
All, please re-review.  This PR now does less than it used to because some of the preceding PRs already accomplished some of the goals of this PR.

File: index.html
Comment (iherman) [inline index.html:2074]: Why removing this?

The preceding paragraph on enveloping proof uses the same terminology to 'recommend' jose-cose. This paragraph does the same for DI. Either *both* _RECOMMENDED_ statements are removed, or none of them. Personally, I prefer keeping it as is: after all, we are 'recommending' the two proof mechanisms that this WG standardizes, which is perfectly fine, and it certainly does not exclude third parties coming up with their own.

File: index.html
Comment (iherman) [inline index.html:2324]: I do not think it is necessary for this example. Example 18 does not use _any_ proof mechanisms, so the reference to DI looks irrelevant.

**Review by iherman — CHANGES_REQUESTED** [review]

File: index.html
Comment (msporny) [reply index.html:2074]: Agreed. Do not remove this, the WG is explicitly recommending at least the securing mechanisms that it is publishing as standards.

**Review by msporny — CHANGES_REQUESTED** [review]
Just a few blocking requests, after these are fixed, we can merge:

* Do not remove the recommendation of using the securing mechanisms the WG is creating.
* Do not remove text related to DI and replace it with text related to JWT -- either talk about both mechanisms specifically, or securing mechanisms generally, or be silent.
* Keep the mentions that `proof` or securing data is placed in a separate graph; it's important to be able to separate the securing mechanism from the data being secured.

File: index.html
Comment (msporny) [inline index.html:2839]: We will want to just reword this to refer to the "Securing Mechanisms" section. This is not blocking feedback, and it's editorial, but while we're in here we might as well change this/update it to point to a better place.

File: index.html
Comment (msporny) [inline index.html:3590]:
```suggestion
information to the securing mechanism utilized. The `evidence`
```

File: index.html
Comment (msporny) [inline index.html:3597]:
```suggestion
information about securing mechanisms, see Section
<a href="#securing-mechanisms"></a>.
```

Comment (msporny) [inline index.html:3597]:
```suggestion
information about securing mechanisms, see Section
<a href="#securing-mechanisms"></a>.
```

File: index.html
Comment (msporny) [inline index.html:4067]: The `controller` DT needs to be deleted here as well.

As a note to the VCWG, @selfissued checked with @awoie (who is the one that raised the issue that resulted in this text) to see if he was ok with removing this text, and he was.

File: index.html
Comment (msporny) [inline index.html:4986]:
```suggestion
The contents of a [=credential=] are secured using a securing mechanism.
```

File: index.html
Comment (msporny) [inline index.html:4987]:
```suggestion
Values used to represent the securing mechanism
```

File: index.html
Comment (msporny) [inline index.html:6501]: -1 to removing this text AND adding the JWT text. Either remove all of it, or add all of it.

File: terms.html
Comment (msporny) [inline terms.html:119]: @iherman this is removing all notions that `proof` is a separate graph, which doesn't seem like a good idea. You added this text, thoughts?

Comment (iherman) [reply terms.html:119]: Good point but, with this new setup, the fact that a `proof` create a new named graph must be added somewhere to the DI spec, and it is ok not to have it there.

At this point we probably should raise a new issue in the DI repo to remind us that this MUST be done, and leave this change as is.

File: index.html
Comment (brentzundel) [reply index.html:2074]: both paragraphs should be consistent

File: index.html
Comment (brentzundel) [inline index.html:6490]:
```suggestion
Any additional requirements defined by the securing mechanism are satisfied.
```

File: index.html
Comment (TallTed) [inline index.html:3589]: This is the first part of one suggestion --
```suggestion
The `evidence` [=property=] provides information that is different from and
```

File: index.html
Comment (TallTed) [reply index.html:3590]: This is the second (and last) part of one suggestion --
```suggestion
complementary to that provided by the securing mechanism in use. The `evidence`
```

File: index.html
Comment (TallTed) [inline index.html:3593]: Since the rest of this paragraph is comparing the `evidence` property to the securing mechanism, that should also happen here.
```suggestion
contrast, the securing mechanism is used to express
```

File: index.html
Comment (TallTed) [inline index.html:4281]:
```suggestion
container</a>. This requires the creation of [=named graphs=], used to isolate
```

File: index.html
Comment (TallTed) [inline index.html:6501]: The new line belongs with the preceding (retained) paragraph. The undeleted lines should be a distinct (subsequent) paragraph.
```suggestion
The JWT <code>iat</code> claim likewise provides the time that the signature was made.
<br/><br/>
The `verificationMethod` [=property=] specifies, for example, the
public key that can be used to verify the digital signature. Dereferencing a
public key URL reveals information about the controller of the key, which can
be checked against the issuer of the [=credential=]. The
`proofPurpose` [=property=] clearly expresses the purpose for
the proof and ensures this information is protected by the signature. A proof is
typically attached to a [=verifiable presentation=] for authentication
purposes and to a [=verifiable credential=] as a method of assertion.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

File: index.html
Comment (selfissued) [inline index.html:2074]:
```suggestion
included in the serialization of the data model. One such RECOMMENDED embedded
```

File: index.html
Comment (selfissued) [inline index.html:2324]:
```suggestion
        <pre class="example nohighlight" title="Basic structure of a presentation">
```

File: index.html
Comment (msporny) [inline index.html:2837]:
```suggestion
Secure the [=credential=] with a <a href="#securing-mechanisms">securing mechanism</a> establishing that the
```

File: index.html
Comment (selfissued) [inline index.html:4047]:
```suggestion
```

File: index.html
Comment (selfissued) [reply index.html:4067]: `controller` now also deleted

File: terms.html
Comment (selfissued) [reply terms.html:119]: I agree with @iherman that this information belongs in DI - not VCDM.

**Review by msporny — APPROVED** [review]

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.

**Comment by TallTed** [comment]
The [history of this PR](https://github.com/w3c/vc-data-model/pull/1397#commits-pushed-5604b06) *suggests* that all my requested changes were applied, but they were NEITHER ACCURATELY NOR COMPLETELY applied, apparently because the "commit" buttons provided by GitHub were not used on all (if any) suggestions.

Manual application of change requests is HORRIBLY error prone, and it is impossible to know whether requested changes were rejected intentionally (which should have led to discussion and agreed resolution) or accidentally — which means manual review of each-and-every change request and subsequent commit must now be performed.

I noticed this problem when skimming the ["files changed"](https://github.com/w3c/vc-data-model/pull/1397/files) after the PR was merged. The word "complementary" was deleted, where it should have moved from one line to the next. The original sentence read —
```
The `evidence` [=property=] provides different and complementary
information to the `proof` [=property=].
```
The remaining sentence is nonsensical — 
```
The `evidence` [=property=] provides information that is different from and
information to the securing mechanism utilized.
```
It *should* now read —
```
The `evidence` [=property=] provides information that is different from and
complementary to that provided by the securing mechanism in use.
```
I am VERY unhappy.

**Comment by TallTed** [comment]
Perhaps GitHub is providing the stress here... The [files changed](https://github.com/w3c/vc-data-model/pull/1397/files) shows the broken sentence above, while the [main branch](https://github.com/selfissued/vc-data-model/blob/5380c8eb4ddb723864be1e6acff239bcdb795d15/index.html#L3600-L3601) shows the correct sentence.

I don't know of a good way to review this, that will accurately show the requested changes and their disposition, other than comparing the PR to the Main branch in side-by-side browser tabs...

**Pull request merged**
